bin/ruby-build

```diff
@@ -712,6 +727,15 @@ build_package_standard() {
     fi
     if [ -z "$CC" ] && is_mac 1010; then
       export CC=clang
+    elif [ "$ruby_semver" -lt 300200 ]; then
+      # GCC v15 changed the default mode from gnu17 to gnu23.
+      # See: https://gcc.gnu.org/gcc-15/changes.html#c
+      # Ruby older than 3.2 can't compile on GCC v15 in the default mode of gnu23.
+      # See: https://github.com/rbenv/ruby-build/discussions/2529
+      if is_fedora && is_fedora 42; then
+        # Fedora 42+ uses GCC v15 with mode gnu23 as default, so we configure for mode gnu17
+        export CFLAGS="$CFLAGS -std=gnu17"
```

Comment:
Would we consider reversing the order of arguments so that user-supplied RUBY_CFLAGS could potentially override the `-std` flag, if necessary?
```suggestion
        export CFLAGS="-std=gnu17 $CFLAGS "
```